pyicloud/services/drive.py

```diff
@@ -53,6 +56,143 @@ def get_file(self, file_id, **kwargs):
+
+    def _update_contentws(self, folder_id, sf_info, document_id, file_object):
+        request = self.session.post(
+            self._document_root + "/ws/com.apple.CloudDocs/update/documents",
+            params=self.params,
+            headers={"Content-Type": "text/plain"},
+            data=json.dumps(
+                {
+                    "data": {
+                        "signature": sf_info["fileChecksum"],
+                        "wrapping_key": sf_info["wrappingKey"],
+                        "reference_signature": sf_info["referenceChecksum"],
+                        "receipt": sf_info["receipt"],
+                        "size": sf_info["size"],
+                    },
+                    "command": "add_file",
+                    "create_short_guid": True,
+                    "document_id": document_id,
+                    "path": {
+                        "starting_document_id": folder_id,
+                        "path": file_object.name,
+                    },
+                    "allow_conflict": True,
+                    "file_flags": {
+                        "is_writable": True,
+                        "is_executable": False,
+                        "is_hidden": False,
+                    },
+                    "mtime": int(time.time()),
+                    "btime": int(time.time()),
+                }
+            ),
+        )
+        if not request.ok:
+            return None
+        return request.json()
+
+    def send_file(self, folder_id, file_object):
+        """Send new file to iCloud Drive."""
+        document_id, content_url = self._get_upload_contentws_url(file_object)
+
+        request = self.session.post(content_url, files={file_object.name: file_object})
+        if not request.ok:
+            return None
+        content_response = request.json()["singleFile"]
+
+        self._update_contentws(folder_id, content_response, document_id, file_object)
+
+    def create_folders(self, parent, name):
+        """Creates a new iCloud Drive folder"""
+        request = self.session.post(
+            self._service_root + "/createFolders",
+            params=self.params,
+            headers={"Content-Type": "text/plain"},
+            data=json.dumps(
+                {
+                    "destinationDrivewsId": parent,
+                    "folders": [
+                        {
+                            "clientId": self.params["clientId"],
+                            "name": name,
+                        }
+                    ],
+                }
+            ),
+        )
+        return request.json()
+
+    def rename_items(self, node_id, etag, name):
+        """Renames an iCloud Drive node"""
+        request = self.session.post(
+            self._service_root + "/renameItems",
+            params=self.params,
+            data=json.dumps(
+                {
+                    "items": [
+                        {
+                            "drivewsid": % node_id,
```

Comment:
`pyicloud/services/drive.py:167:42: E0001: invalid syntax (<unknown>, line 167) (syntax-error)`

Reply:
Ouch. Missed that. Sorry. 🤬